packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3550,9 +3567,104 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     }
   }
 
-  void _onEditableScroll() {
-    _selectionOverlay?.updateForScroll();
-    _scribbleCacheKey = null;
+  final bool _platformSupportsFadeOnScroll = switch (defaultTargetPlatform) {
+    TargetPlatform.android ||
+    TargetPlatform.iOS => true,
+    TargetPlatform.fuchsia ||
+    TargetPlatform.linux ||
+    TargetPlatform.macOS ||
+    TargetPlatform.windows => false,
+  };
+
+  bool _isInternalScrollableNotification(BuildContext? notificationContext) {
+    final ScrollableState? scrollableState = notificationContext?.findAncestorStateOfType<ScrollableState>();
+    if (_scrollableKey.currentContext == scrollableState?.context) {
+      return true;
+    }
+    return false;
+  }
+
+  void _handleContextMenuOnParentScroll(ScrollNotification notification) {
+    if (_isInternalScrollableNotification(notification.context)) {
+      return;
+    }
+    if (!_scrollableNotificationIsFromSameSubtree(notification.context)) {
+      return;
+    }
+    _handleContextMenuOnScroll(notification);
+  }
+
+  Rect _calculateViewportRect() {
+    final Size screenSize = MediaQuery.sizeOf(context);
+    final ui.FlutterView view = View.of(context);
+    final double obscuredVertical = (view.padding.top + view.padding.bottom + view.viewInsets.bottom) / view.devicePixelRatio;
+    final double obscuredHorizontal = (view.padding.left + view.padding.right) / view.devicePixelRatio;
```

Comment:
Do you have other plans to update the implementation here? Using the view dimensions is a bit strange, for example when the viewport is not fullscreen.

Reply:
(Or you could move this to a different PR and always show the edit menu for now)

Reply:
I don't follow, do you mean when the textfield is inside a viewport that is not the fullscreen?

Reply:
My thought with the logic here is:

`selectionIsVisible` tells us if the selection is visible within the `RenderEditable`.
`renderEditableBounds` is the `Rect` for the `RenderEditable`, if it `hasNaN` it is likely not laid out.
`viewportRect` attempts to calculate the device entire viewport with paddings for system overlays.
`renderEditableInView` checks whether the `viewportRect` and `renderEditableBounds` overlap, if they do then the `RenderEditable` should be visible within the device viewport.

The final check is `selectionIsVisible && renderEditableInView` which checks if the selection is visible in the `RenderEditable` and if the `RenderEditable` is visible within the device viewport.

Reply:
I think it makes more sense to hide the overlay if the `RenderEditable` itself is not visible. This way the menu still adjusts to avoid the overlays if the text field is visible and hides it when its completely obscured to prevent the menu from overlapping with the system overlays. This seems consistent with the Android behavior.

Taking system overlays into account:
https://github.com/flutter/flutter/assets/948037/f45c9172-0815-4362-bcfd-01b4af02b2d6

Without taking system overlays into account:
https://github.com/flutter/flutter/assets/948037/4fe00dab-eccb-49fa-8627-ef5a2a65f8a3

Jetpack compose behavior:
https://github.com/flutter/flutter/assets/948037/76ca8bce-dc27-4e78-a8d3-4289b636c413

Reply:
I see the behavior makes sense to me. But the code seems to only check for system UI obstruction, that seems oddly specific. If the text field is obstructed by the outer scroll view would the menu also disappear?

Reply:
Isn't showOnScreen/showInViewport/or ensureVisible called as part of focusing on a text field and showing the context menu? If so, it should account for the case where a nested scroll view is scrolled out of view by the outer scroll view.

Reply:
Yeah my implementation of trying to calculate the viewport rect didn't take into account for this case because the viewport is not always the devices fullscreen. I was able to fix this case by going back to your original suggestion @Piinks https://github.com/flutter/flutter/pull/138313#discussion_r1409994272 and bubbling through the viewports. I still think we should consider the calculated viewport rect for system overlays like the keyboard.